test/integration/auth/node_test.go

```diff
@@ -248,6 +275,9 @@ func TestNodeAuthorizer(t *testing.T) {
 	expectAllowed(t, createNode2(node2Client))
 	expectAllowed(t, updateNode2Status(node2Client))
 	expectAllowed(t, deleteNode2(node2Client))
+	// node evicting pod bound to itself is allowed
+	expectAllowed(t, createNode2MirrorPod(node2Client))
+	expectAllowed(t, createNode2MirrorPodEviction(node2Client))
```

Comment:
Those docs are misleading... they mean you cannot cause a mirror pod to stop running on a kubelet by deleting it from the API (the kubelet would just recreate it), but you can absolutely delete the API object.